infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesSnapshotReader.java

```diff
@@ -75,6 +75,8 @@ public class RoutingQueuesSnapshotReader extends BaseSnapshotReader {
     // Boolean indicating if the Snapshot reader is waiting for a start marker from this snapshot sync
     private boolean waitingForStartMarker = true;
 
+    private OpaqueStream opaqueStream = null;
```

Comment:
if we have task based abstraction then maybe multiple threads can access this? wondering if we should make it `volatile` ?